CMakeLists.txt

```diff
@@ -836,6 +836,12 @@ zephyr_get_include_directories_for_lang(C
   STRIP_PREFIX # Don't use a -I prefix
   )
 
+if(CONFIG_PM_DEVICE_POWER_DOMAIN_DYNAMIC)
```

Comment:
We should introduce a hidden Kconfig option for the CONFIG_DYNAMIC_DEVICES that gets selected by CONFIG_PM_DEVICE_POWER_DOMAIN_DYNAMIC.  So that the dynamic device functionality could be used by other use cases.

Reply:
That makes sense

Reply:
I'll just change a little bit it, `CONFIG_PM_DEVICE_POWER_DOMAIN_DYNAMIC` contains the number of slots that should be pre-allocated, since we don't have the need to resize all devices but only power domains, I'll keep the power domain name. What I'll change though is the hidden option that tells if the build supports dynamic devices. In the end this is the symbol that is used inside `device.h` and linker scripts. If I remember correctly the concern was about having PM specifics in those files.

The other possibility is to make it generic and resize all devices.

Reply:
the idea was to guard with CONFIG_DYNAMIC_DEVICES (which would be selected by CONFIG_PM_DEVICE_POWER_DOMAIN_DYNAMIC > 0 ) and then use CONFIG_PM_DEVICE_POWER_DOMAIN_DYNAMIC below now for power domain, but someone else might add more devices, so they feature is exposed as a generic one. Really cosmetic and can be done later when there are more users, but I guess this would make it cleaner.

Reply:
that is basically what I've done AFAIU, just the name that is different.